grails-app/controllers/au/org/ala/layers/LayerController.groovy

```diff
@@ -353,44 +354,48 @@ class LayerController {
 
     }
 
+    @RequireAdmin
     def download(String id) {
         Layer l = layerDao.getLayerByDisplayName(id)
-        if (downloadAllowed(l)) {
-            OutputStream outputStream = null
-            try {
-                outputStream = response.outputStream as OutputStream
-                //write resource
-                response.setContentType("application/octet-stream")
-                response.setHeader("Content-disposition", "attachment;filename=${id}.zip")
-
-                // When a geotiff exists, only download the geotiff
-                def path = "/layer/${l.name}"
-                def geotiff = new File(grailsApplication.config.data.dir + path + ".tif")
-                if (geotiff.exists()) {
-                    path += ".tif"
-                }
+        if(l){
```

Comment:
single letter variables in methods are not encouraged - makes code hard to read. They are acceptable inside closures only.
Edit: I realise it was already like this, but if you touch code then try and improve it is possible.

Reply:
ALA code style!!!